node/src/actors/chain_manager/mining.rs

```diff
@@ -705,9 +705,19 @@ fn build_block(
     }
 
     for ta_tx in tally_transactions {
-        if let Some(dr_output) = dr_pool.get_dr_output(&ta_tx.dr_pointer) {
+        if let Some(dr_state) = dr_pool.data_request_state(&ta_tx.dr_pointer) {
             tally_txns.push(ta_tx.clone());
-            transaction_fees += dr_output.tally_fee;
+            let commits_count = dr_state.info.commits.len();
```

Comment:
This repeated logic makes me think that `calculate_witness_reward` should be a method of `dr_state`, so we can just do `dr_state.calculate_witness_reward(&dr_pointer)`

Reply:
honests_count can not be calculated with dr_state information

Reply:
Then what about `dr_state.calculate_witness_reward(&dr_pointer, &tally_transaction)`

Reply:
There are another functions where is used (i.e. create_tally), where do not have access to dr_state